internal/pkg/service/appproxy/http/router.go

```diff
@@ -131,6 +131,10 @@ func (r *Router) createDataAppHandler(ctx context.Context, app DataApp) http.Han
 
 	mux := http.NewServeMux()
 
+	// Always send /_proxy/ requests to the correct provider.
+	// This is necessary for proxy callback url to work on an app with prefixed private parts.
+	mux.Handle("/_proxy/", r.createMultiProviderHandler(oauthProviders))
```

Comment:
The `multiProviderHandler` is created multiples times, I think that this could be avoided, and we can check whether the provider has access to the path in the a new `UpstreamHandler', but let's leave it for now.